content/tutorials/use-specialized-payment-types/create-custom-transactors.md

```diff
@@ -0,0 +1,379 @@
+        sleCheck->setFieldU32(sfSourceTag, *srcTag);
+    if (auto const dstTag = ctx_.tx[~sfDestinationTag])
+        sleCheck->setFieldU32(sfDestinationTag, *dstTag);
+    if (auto const invoiceId = ctx_.tx[~sfInvoiceID])
+        sleCheck->setFieldH256(sfInvoiceID, *invoiceId);
+    if (auto const expiry = ctx_.tx[~sfExpiration])
+        sleCheck->setFieldU32(sfExpiration, *expiry);
+
+    view().insert(sleCheck);
+
+    auto viewJ = ctx_.app.journal("View");
+    // If it's not a self-send (and it shouldn't be), add Check to the
+    // destination's owner directory.
+    if (dstAccountId != account_)
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(dstAccountId),
+            checkKeylet,
+            describeOwnerDir(dstAccountId));
+
+        JLOG(j_.trace()) << "Adding Check to destination directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfDestinationNode, *page);
+    }
+
+    {
+        auto const page = view().dirInsert(
+            keylet::ownerDir(account_),
+            checkKeylet,
+            describeOwnerDir(account_));
+
+        JLOG(j_.trace()) << "Adding Check to owner directory "
+                        << to_string(checkKeylet.key) << ": "
+                        << (page ? "success" : "failure");
+
+        if (!page)
+            return tecDIR_FULL;
+
+        sleCheck->setFieldU64(sfOwnerNode, *page);
+    }
+    // If we succeeded, the new entry counts against the creator's reserve.
+    adjustOwnerCount(view(), sle, 1, viewJ);
+    return tesSUCCESS;
+}
+```
+
+
+## Additional Functions
+
+
+### `calculateBaseFee`
+
+The `calculateBaseFee` function calculates the XRP fee to run a transaction. The fee is usually inherited from the base transactor, but transactors such as `EscrowFinish` require additional fees.
+
+```c++
+XRPAmount
+EscrowFinish::calculateBaseFee(ReadView const& view, STTx const& tx)
+{
+    XRPAmount extraFee{0};
+
+    if (auto const fb = tx[~sfFulfillment])
+    {
+        extraFee += view.fees().base * (32 + (fb->size() / 16));
+    }
+
+    return Transactor::calculateBaseFee(view, tx) + extraFee;
+}
+```
+
+
+### `makeTxConsequences`
+
+`rippled` uses a [`TxConsequences`](https://github.com/XRPLF/rippled/blob/master/src/ripple/app/tx/applySteps.h#L41-L44) class to describe the outcome to an account when applying a transaction. It tracks the fee, maximum possible XRP spent, and how many sequences are consumed by the transaction. There are three types of consequences:
+
+- **Normal:** The transactor doesn't affect transaction signing and only consumes an XRP fee.
+- **Blocker:** The transactor affects transaction signing, preventing valid transactions from queueing behind it.
+- **Custom:** The transactor needs to do additional work to determination consequences.
```

Comment:
You should probably clarify that any transaction that can _spend_ XRP (beyond the basic transaction cost) needs a custom `makeTxConsequences`. The [PaymentChannelCreate transaction's example](https://github.com/XRPLF/rippled/blob/aded4a7a925a5822522e6bbbb4adfd3fec39257b/src/ripple/app/tx/impl/PayChan.cpp#L165-L169) is a nice small one. (It can spend up to the `Amount` field of the transaction).